src/Microsoft.DotNet.XHarness.CLI/Commands/WASI/Engine/WasiTestCommand.cs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasi;
+using Microsoft.DotNet.XHarness.CLI.Commands.Wasm;
+using Microsoft.DotNet.XHarness.Common;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.Execution;
+using Microsoft.DotNet.XHarness.Common.Logging;
+using Microsoft.DotNet.XHarness.Common.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Wasi;
+internal class WasiTestCommand : XHarnessCommand<WasiTestCommandArguments>
+{
+    private const string CommandHelp = "Executes tests on WASI using a selected engine";
+
+    protected override WasiTestCommandArguments Arguments { get; } = new();
+    protected override string CommandUsage { get; } = "wasi test [OPTIONS] -- [ENGINE OPTIONS]";
+    protected override string CommandDescription { get; } = CommandHelp;
+
+    public WasiTestCommand() : base(TargetPlatform.WASI, "test", true, new ServiceCollection(), CommandHelp)
+    {
+    }
+
+    protected override async Task<ExitCode> InvokeInternal(ILogger logger)
+    {
+        var processManager = ProcessManagerFactory.CreateProcessManager();
+
+        string engineBinary = Arguments.Engine.Value switch
+        {
+            WasmEngine.WasmTime => "wasmtime",
+            _ => throw new ArgumentException("Engine not set")
+        };
+
+        if (!string.IsNullOrEmpty(Arguments.EnginePath.Value))
+        {
+            engineBinary = Arguments.EnginePath.Value;
+            if (Path.IsPathRooted(engineBinary) && !File.Exists(engineBinary))
+                throw new ArgumentException($"Could not find js engine at the specified path - {engineBinary}");
+        }
+        else
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {                
+                engineBinary = FileUtils.FindFileInPath(engineBinary + ".cmd");
+            }
+        }
+      
+        logger.LogInformation($"Using wasm engine {Arguments.Engine.Value} from path {engineBinary}");
+        await PrintVersionAsync(Arguments.Engine.Value.Value, engineBinary);
```

Comment:
Move these inside the try/catch, so a missing engine error can be reported correctly. Do the same for `WasmTestCommand`.